### PR TITLE
DOC: add example to insert (#39313)

### LINKED ISSUE #39313: DOC: add examples to DataFrame.insert()
#### Code Sample

```
import pandas as pd
#create a df and insert a series astype String
df = pd.DataFrame({1:[1,2,3]},index=[0,1,3])
df.insert(loc=0,column='test',value=pd.Series(["one","two","three"]).astype("string"))
print(df)
#The values inserted normally evaluates correct though
print('==============')
print(pd.Series(["one","two","three"]).astype("string"))
```

Output:
```
   test  1
0   one  1
1   two  2
3  <NA>  3

==============

0      one
1      two
2    three
dtype: string

```

#### Problem description
Issue: StringDtype exhibits strange behaviour when inserting as column. Some elements are inserted as <NaN>, but evaluates correctly outside of the insert function

#### Expected Output

Values should be inserted as is.

```
   test  1
0   one  1
1   two  2
3  three  3

==============

0      one
1      two
2    three
dtype: string

```
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 3e89b4c4b1580aa890023fc550774e63d499da25
python           : 3.8.3.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.18.0-240.1.1.el8_3.x86_64
Version          : #1 SMP Thu Nov 19 17:20:08 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_ZA.UTF-8
LOCALE           : en_ZA.UTF-8

pandas           : 1.2.0
numpy            : 1.19.5
pytz             : 2020.5
dateutil         : 2.8.1
pip              : 20.3.3
setuptools       : 49.6.0
Cython           : 0.29.21
pytest           : 6.2.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.2
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.19.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.3
numexpr          : 2.7.2
odfpy            : None
openpyxl         : 3.0.6
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.6.0
sqlalchemy       : 1.3.22
tables           : 3.6.1
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

#### Workaround

Parse to "Values" before inserting. In the example above:

df.insert(loc=0,column='test',value=pd.Series(["one","two","three"]).astype("string").values)

### PR DESCRIPTION
**Pull request by nofarm3**
- [x] closes #39313
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry - no

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Thanks @nofarm3 - the example looks good, but I get a couple of warnings when trying to build it locally (`python make.py --single pandas.DataFrame.insert`):
```
reading sources... [100%] reference/api/pandas.DataFrame.insert                                                                                                                                           
/home/marco/pandas-marco/pandas/core/frame.py:docstring of pandas.core.frame.DataFrame.insert:58: WARNING: Unexpected indentation.
/home/marco/pandas-marco/pandas/core/frame.py:docstring of pandas.core.frame.DataFrame.insert:59: WARNING: Block quote ends without a blank line; unexpected unindent.
```

**Comment by nofarm3** [comment]
@MarcoGorelli Thanks Marco, fixed.
I think the current failures are not related to my change.

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me, thanks @nofarm3 !

File: pandas/core/frame.py [hunk 1]
```diff
@@ -3870,6 +3870,14 @@ def insert(self, loc, column, value, allow_duplicates: bool = False) -> None:
            col1  col1  newcol  col2
         0   100     1      99     3
         1   100     2      99     4
+
+        Notice that Pandas uses index alignment in case of `value` from type `Series`:
```
Comment (MarcoGorelli) [inline pandas/core/frame.py:3874]:
```suggestion
        Notice that pandas uses index alignment in case of `value` from type `Series`:
```

**Pull request merged**